pytest/conftest.py

```diff
@@ -0,0 +1,90 @@
+                                                              'end': '12/12/2010'}}})
+    statement = 'This resource is shared under the Creative Commons Attribution CC BY.'
+    url = 'http://creativecommons.org/licenses/by/4.0/'
+    metadata.append({'rights': {'statement': statement, 'url': url}})
+    metadata.append({'language': {'code': 'fre'}})
+
+    # contributor
+    con_name = 'Mike Sundar'
+    con_org = "USU"
+    con_email = 'mike.sundar@usu.edu'
+    con_address = "11 River Drive, Logan UT-84321, USA"
+    con_phone = '435-567-0989'
+    con_homepage = 'http://usu.edu/homepage/001'
+    con_identifiers = {'ORCID': 'https://orcid.org/mike_s',
+                       'ResearchGateID': 'https://www.researchgate.net/mike_s'}
+    metadata.append({'contributor': {'name': con_name,
+                                     'organization': con_org, 'email': con_email,
+                                     'address': con_address, 'phone': con_phone,
+                                     'homepage': con_homepage,
+                                     'identifiers': con_identifiers}})
+
+    # creator
+    cr_name = 'John Smith'
+    cr_org = "USU"
+    cr_email = 'jsmith@gmail.com'
+    cr_address = "101 Clarson Ave, Provo UT-84321, USA"
+    cr_phone = '801-567=9090'
+    cr_homepage = 'http://byu.edu/homepage/002'
+    cr_identifiers = {'ORCID': 'https://orcid.org/john_smith',
+                      'ResearchGateID': 'https://www.researchgate.net/john_smith'}
+    metadata.append({'creator': {'name': cr_name, 'organization': cr_org,
+                                 'email': cr_email, 'address': cr_address,
+                                 'phone': cr_phone, 'homepage': cr_homepage,
+                                 'identifiers': cr_identifiers}})
+
+    # relation
+    metadata.append({'relation': {'type': 'isPartOf',
+                                  'value': 'http://hydroshare.org/resource/001'}})
+    # source
+    metadata.append({'source': {'derived_from': 'http://hydroshare.org/resource/0001'}})
+
+    # identifier
+    metadata.append({'identifier': {'name': 'someIdentifier', 'url': 'http://some.org/001'}})
+
+    # fundingagency
+    agency_name = 'NSF'
+    award_title = "Cyber Infrastructure"
+    award_number = "NSF-101-20-6789"
+    agency_url = "http://www.nsf.gov"
+    metadata.append({'fundingagency': {'agency_name': agency_name, 'award_title': award_title,
+                                       'award_number': award_number, 'agency_url': agency_url}})
+
+    user = User.objects.get(username='admin')
+
+    user_access = UserAccess(user=user)
+    user_access.save()
+    user_labels = UserLabels(user=user)
+    user_labels.save()
+
+    metadata = json.loads(json.dumps(metadata))
+
+    _res = hydroshare.create_resource(
+        resource_type=rtype,
+        owner=user,
+        title=title,
+        metadata=metadata,
+        files=(open('assets/cea.tif'),)
+    )
+    yield res_uuid
+    _res.delete()
```

Comment:
How does the user object that is created in this function be cleaned up between different tests that use this function?

Reply:
the admin user is the admin:default user and is pre-existing in the Django framework/our configs. I never created the admin user. I did, however modify it to allow it to interact with resources, as by default the admin user can't do anything other than interact with the admin panel

Reply:
@cuahsimarko I'd suggest to create a test user for functional testing rather than to use the default admin user since admin user may have more privilege than a regular user.

Reply:
Using the admin user is to use the admin_client and not get bogged down for authentication. The functional tests for authed users can be in Selenium, nose... or a specific test for a particular, authenticated user can be written later.

Reply:
@cuahsimarko OK. Thanks for the clarification. Then what specific functionality does using this admin user test? Can it be covered by tests using an authenticated test user?

Reply:
A lot of additional work can occur in a separate PR to build out what is needed to create a properly authenticated user (and automatically teardown/cleanup) and write a sample test that uses this.